airtouch4pyapi/packetmap.py

```diff
@@ -1,133 +1,302 @@
+            if(SettingValueTranslator.map[atVersion][attributeName][namedValue] == rawValue):
                 return namedValue;
         return ""
 
 
 class DataLocationTranslator:
     map = {
-        "GroupStatus" : {
-            "PowerState" : "1:7-8",
-            "GroupNumber" : "1:1-6",
-            "ControlMethod" : "2:8-8",
-            "OpenPercentage" : "2:1-7",
-            "BatteryLow" : "3:8-8",
-            "TurboSupport" : "3:7-7",
-            "TargetSetpoint" : "3:1-6",
-            "Sensor" : "4:8-8",
-            "Temperature" : "5:6-16",
-            "Spill": "6:5-5"
-        },
-        "AcStatus" : {
-            "PowerState" : "1:7-8",
-            "AcNumber" : "1:1-6",
-            "AcMode" : "2:5-8",
-            "AcFanSpeed" : "2:1-4",
-            "Spill" : "3:8-8",
-            "AcTimer" : "3:7-7",
-            "AcTargetSetpoint" : "3:1-6",
-            "Temperature" : "5:6-16",
-        },
-        "GroupControl" : {
-            "GroupNumber" : "1:1-8",
-            "GroupSettingValue" : "2:6-8",
-            "HaveTemperatureControl" : "2:4-5",
-            "Power" : "2:1-3",
-            "TargetSetpoint" : "3:1-8",
-            "ZeroedByte" : "4:1-8"
+        4 : {
+            "GroupStatus" : {
+                "PowerState" : "1:7-8",
+                "GroupNumber" : "1:1-6",
+                "ControlMethod" : "2:8-8",
+                "OpenPercent" : "2:1-7",
+                "BatteryLow" : "3:8-8",
+                "TurboSupport" : "3:7-7",
+                "TargetSetpoint" : "3:1-6",
+                "Sensor" : "4:8-8",
+                "Temperature" : "5:6-16",
+                "Spill": "6:5-5"
+            },
+            "AcStatus" : {
+                "PowerState" : "1:7-8",
+                "AcNumber" : "1:1-6",
+                "AcMode" : "2:5-8",
+                "AcFanSpeed" : "2:1-4",
+                "Spill" : "3:8-8",
+                "AcTimer" : "3:7-7",
+                "AcTargetSetpoint" : "3:1-6",
+                "Temperature" : "5:6-16",
+            },
+            "GroupControl" : {
+                "GroupNumber" : "1:1-8",
+                "GroupSettingValue" : "2:6-8",
+                "HaveTemperatureControl" : "2:4-5",
+                "Power" : "2:1-3",
+                "TargetSetpoint" : "3:1-8",
+                "ZeroedByte" : "4:1-8"
+            },
+            "AcControl" : {
+                "Power" : "1:7-8",
+                "AcNumber" : "1:1-6",
+                "AcMode" : "2:5-8",
+                "AcFanSpeed" : "2:1-4",
+                "SetpointControlType" : "3:7-8",
+                "TargetSetpoint" : "3:1-6",
+                "ZeroedByte" : "4:1-8"
+            },
+            "AcAbility" : {
+                #byte number - 2 from the spec, due to fixed message at start
+                "AcNumber" : "1:1-8",
+                "ChunkSize" : "2:1-8",
+                "AcName" : "3:1-128",
```

Comment:
I wish that this worked! but it doesn't - we will have to remove this mapping from here for now and revert to the manual AcName placeholder specified elsewhere for AT 4

Reply:
OK!